source-sqlserver/prerequisites.go

```diff
@@ -232,11 +233,53 @@ func (db *sqlserverDatabase) prerequisiteTableCaptureInstance(ctx context.Contex
 	}
 
 	// Otherwise we attempt to create one
-	const query = `EXEC sys.sp_cdc_enable_table @source_schema = @p1, @source_name = @p2, @role_name = @p3, @capture_instance = @p4;`
-	var instanceName = fmt.Sprintf("%s_%s", schema, table)
-	if _, err := db.conn.ExecContext(ctx, query, schema, table, db.config.User, instanceName); err == nil {
+	if instanceName, err := cdcCreateCaptureInstance(ctx, db.conn, schema, table, db.config.User); err == nil {
 		logEntry.WithField("instance", instanceName).Info("enabled cdc for table")
 		return nil
 	}
 	return fmt.Errorf("table %q has no capture instances and user %q cannot create one", streamID, db.config.User)
 }
+
+func cdcCreateCaptureInstance(ctx context.Context, conn *sql.DB, schema, table, username string) (string, error) {
+	// SQL Server table names may be up to 128 characters, but capture instance names must
+	// be at most 100 characters and we have other information to cram in there. The names
+	// must be unique, but other than that they might as well be random strings, the logic
+	// around capture instance selection is based on the (schema, table, instance) info in
+	// `cdc.change_tables` and doesn't care about the names.
+	//
+	// So this logic is just trying to give them a reasonably unique name which also tells
+	// us some useful facts, like as much of the name as we can accomodate and when it was
+	// created. The timestamp also helps ensure that successive capture instances from the
+	// same table are always unique.
+
+	// Human-readable prefix of the name, truncated to 64 characters
+	var prefix = schema + "_" + table
+	if len(prefix) > 64 {
+		prefix = prefix[:64]
```

Comment:
That's a good catch. I think the answer is that yes, it is theoretically possible to create a database where this would cause problems (or at least where we truncate a multi-byte character into an incorrect code point -- it's also possible that this would be translated into the Unicode replacement character and everything would still work correctly even then). However doing this would require setting the collation to a non-default value at database creation time, since the default is `SQL_Latin1_General_CP1_CI_AS` which shouldn't have any issues.

I'm not sure if it's worth the effort of fixing this right now. I'll take a look at the Unicode-aware string functions to see if it's reasonably simple to do a proper UTF-8 truncate, but to hit this a user would need to:
1. Modify their server or database-level collation to something that permits multi-byte characters.
2. Actually use multi-byte characters frequently in their table names.
3. Have a really long table name such that we're forced to trim it.
4. Get unlucky about exactly where that trimming point falls.

And if this did happen they'd hit it the very first time they tried to publish, and could work around it temporarily by going and manually creating a capture instance (buying us time to go fix it). So I'll see about fixing this if it looks easy, and otherwise I'll add a comment and not bother just yet.